src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1931,7 +1922,7 @@ internal void MarkStaticConstructorVisibleToReflection (TypeDefinition type, in
 			MarkGenericParameterProvider (type);
 
 			// There are a number of markings we can defer until later when we know it's possible a reference type could be instantiated
-			// For example, if no instance of a type exist, then we don't need to mark the interfaces on that type
+			// For example, if no instance of a type exist, then we don't need to mark the interfaces on that type -- Note this is not true for static interfaces
```

Comment:
While the added comment is correct, I think it makes this whole section confusing. Maybe you should add that the `MarkRequiermentsForInstantiatedTypes` also works for static interfaces - and maybe we should rename that method then.